components/log-viewer-webui/server/src/main.js

```diff
@@ -0,0 +1,65 @@
+import dotenv from "dotenv";
+import process from "node:process";
+
+import app from "./app.js";
+
+
+/**
+ * Parses environment variables into config values for the application.
+ *
+ * @return {{PORT: string, HOST: string}}
+ * @throws {Error} if any required environment variables is undefined.
+ */
+const parseEnvVars = () => {
+    dotenv.config({
+        path: ".env",
+    });
+
+    const {
+        HOST, PORT,
+    } = process.env;
+    const envVars = {
+        HOST, PORT,
+    };
+
+    // Check for mandatory environment variables
+    for (const [key, value] of Object.entries(envVars)) {
+        // eslint-disable-next-line no-undefined
+        if (undefined === value) {
+            throw new Error(`Environment variable ${key} must be defined.`);
+        }
+    }
+
+    return envVars;
+};
+
+/**
+ * Sets up and runs the server.
+ */
+const main = async () => {
+    const envToLogger = {
+        development: {
+            transport: {
+                target: "pino-pretty",
+            },
+        },
+        production: true,
+        test: false,
+    };
+    const server = await app({
+        logger: envToLogger[process.env.NODE_ENV] ?? true,
+    });
+
+    try {
+        const envVars = parseEnvVars();
+        await server.listen({host: envVars.HOST, port: parseInt(envVars.PORT, 10)});
```

Comment:
Do we want to use `Number()` for string-to-decimal conversions? 

Not sure if we had this discussion before. I'm at fault introducing this `parseInt` convention. There's no difference between the two approaches except:
1. Number() requires the arguments to be strictly a number, while `parseInt(,10)` attempt to parse an int from the argument with best effort. e.g. Number("123abc") -> NaN; parseInt("123abc", 10) -> 123.
2. For an empty string, Number("") returns 0, whereas parseInt("", 10) returns NaN.

```suggestion
        await server.listen({host: envVars.HOST, port: Number(envVars.PORT)});
```

Reply:
Ah sorry, forgot about `Number` amongst all the Fastify stuff.